serial/serialutil.py

```diff
@@ -239,9 +239,9 @@ def __init__(self,
             self.inter_byte_timeout = kwargs.pop('interCharTimeout')
         if kwargs:
             raise ValueError('unexpected keyword arguments: {!r}'.format(kwargs))
-
-        if port is not None:
-            self.open()
+        # self.open() shouldn't be here, The open function in rfc2217.py ,so it should be in Serial
```

Comment:
You seem to misunderstand how inheritence works in object-oriented programming.

- `serial.rfc2217.Serial.__init__`  calls `super(Serial, self).__init__()` which results in this function, `SerialBase.__init__`.
- `SerialBase.__init__()` calls `self.open()`
  - `self` still refers to a `serial.rfc2217.Serial` object, so `open` (due to method resolution order) refers to `serial.rfc2217.Serial.open()` (the function you want it to call).

If you remove this call to `self.open()`, how would any of the other classes derived from `SerialBase` actually open their serial ports?

Reply:
I think you can review the code. In fact, this method is just only used in The son class,OK?

Reply:
Besides, if the parent class is instantiated and the self.open method is called, it will result in an error